File: swift/Sources/benchmarks/Base64Decode.swift
import Foundation

final class Base64Decode: BenchmarkProtocol {
  private var sizeVal: Int64 = 0
  private var str2: String = ""
  private var bytes: [UInt8] = []
  private var resultVal: UInt32 = 0

  init() {
    sizeVal = configValue("size") ?? 0
  }

  func prepare() {
    let str = String(repeating: "a", count: Int(sizeVal))
    let data = str.data(using: .utf8)!

    str2 = data.base64EncodedString()
  }

  func run(iterationId: Int) {
    if let data = Data(base64Encoded: str2) {
      bytes = [UInt8](data)
      resultVal &+= UInt32(data.count)
    }
  }

  var checksum: UInt32 {
    let str3 = String(bytes: bytes.prefix(5), encoding: .utf8) ?? "-"
    let truncatedStr2 = str2.count > 4 ? String(str2.prefix(4)) + "..." : str2
    let truncatedStr3 = str3.count > 4 ? String(str3.prefix(4)) + "..." : str3
    let message = "decode \(truncatedStr2) to \(truncatedStr3): \(resultVal)"
    return Helper.checksum(message)
  }

  func name() -> String {
    return "Base64::Decode"
  }
}

File: csharp/Base64Decode.cs
using System.Text;

public class Base64Decode : Benchmark
{
    private int _n;
    private string _str2 = "";
    private byte[] _str3bytes;
    private uint _result;

    public Base64Decode()
    {
        _result = 0;
        _n = (int)ConfigVal("size");
    }

    public override void Prepare()
    {
        string str = new string('a', _n);
        byte[] bytes = Encoding.UTF8.GetBytes(str);
        _str2 = Convert.ToBase64String(bytes);
        _str3bytes = Convert.FromBase64String(_str2);
    }

    public override void Run(long IterationId)
    {
        _str3bytes = Convert.FromBase64String(_str2);
        _result += (uint)_str3bytes.Length;
    }

    public override uint Checksum
    {
        get
        {
            string _str3 = Encoding.UTF8.GetString(_str3bytes);
            string resultStr = $"decode {(_str2.Length > 4 ? _str2.Substring(0, 4) + "..." : _str2)} to {(_str3bytes.Length > 4 ? _str3.Substring(0, 4) + "..." : _str3)}: {_result}";
            return Helper.Checksum(resultStr);
        }
    }
}